nanoFramework.Tools.DebugLibrary.Shared/Extensions/FlashSectorDataExtensions.cs

```diff
@@ -27,12 +27,21 @@ public static DeploymentSector ToDeploymentSector(this FlashSectorData value)
                 int programmingAlignment = 0;
 
                 // check alignment requirements
-                if ((value.Flags
-                     & BlockRegionAttributes_MASK
-                     & BlockRegionAttribute_ProgramWidthIs64bits) == BlockRegionAttribute_ProgramWidthIs64bits)
+                uint blockRegionAttributes = value.Flags & BlockRegionAttributes_MASK;
+                switch (blockRegionAttributes)
                 {
-                    // programming width is 64bits => 8 bytes
-                    programmingAlignment = 8;
+                    case BlockRegionAttribute_ProgramWidthIs64bits:
+                        programmingAlignment = 64 / 8;
+                        break;
+                    case BlockRegionAttribute_ProgramWidthIs128bits:
+                        programmingAlignment = 128 / 8;
+                        break;
+                    case BlockRegionAttribute_ProgramWidthIs256bits:
+                        programmingAlignment = 256 / 8;
+                        break;
+                    default:
```

Comment:
Not very fond of this default here... 🤔 
If the masked value from the flags is not one of the implemented ones, an exception should be thrown. This will caught targets that may have different or new flags, instead of defaulting to 0 (which is already set at `programmingAlignment` declaration).

Reply:
A better implmentation would have it so there were not independent bit flags but a number value.

There is no exception if a bit is set or not set, but there is an exception if more that 1 bit is set.
I couldn't change this easily as it would have impacted on the 64bit flash STM32F4 ?, which I don't have and can't test.

Reply:
We can only make breaking changes if there is an absolute need for it.
In this case, I don't think there is one.
These flags are perfectly isolated, identifiable and produce known values.

In the future, when we add new alignments, and the VS extension is not up to date - when using the code above - we would be flashing with the default alignment which would be wrong and doomed to fail.

So let's not `default` to no alignment please. If you have a suggestion other than throwing an exception for unknown values, lets discuss it.

Reply:
Hi I need some help here.
Maybe its my misunderstaning on what you are asking for.
The previous code always used 0 as the default or 64 bits if bit flag set.
I have added another two flags for 128 and 256, leaving everything else untouched so as not to break existing setups.

As you can see from the previous code below, 0 is the default and a valid response, and no exceptions were checked for.

Happy to try and code something different.
But no flags set, means no alignment required.

        public static DeploymentSector ToDeploymentSector(this FlashSectorData value)
        {
            // build a DeploymentSector from a FlashSectorData

            List<DeploymentBlock> blocks = new List<DeploymentBlock>();

            for (int i = 0; i < value.NumBlocks; i++)
            {
                int programmingAlignment = 0;

                // check alignment requirements
                if ((value.Flags
                     & BlockRegionAttributes_MASK
                     & BlockRegionAttribute_ProgramWidthIs64bits) == BlockRegionAttribute_ProgramWidthIs64bits)
                {
                    // programming width is 64bits => 8 bytes
                    programmingAlignment = 8;
                }

                blocks.Add(new DeploymentBlock(
                    (int)value.StartAddress + (i * (int)value.BytesPerBlock),
                    (int)value.BytesPerBlock,
                    programmingAlignment));
            }

            return new DeploymentSector(blocks);
        }

Reply:
Thanks.
I've just added the check and exception I've been suggesting.